### PR TITLE
Updated CMakeLists with install targets

### PR DESCRIPTION
**Pull request by shaun-edwards**
Fix for issue ros-industrial/ros_industrial_issues#12.

@gavanderhoorn, please review (getting ready for release).

### CODE REVIEW

**Comment by gavanderhoorn** [comment]
See [ros-industrial/industrial_core#58](https://github.com/ros-industrial/industrial_core/issues/58#issuecomment-31631250) for some of my comments.

**Comment by shaun-edwards** [comment]
I don't want the source install to hold up the release.  This is something that can wait until Indigo or be released in a patch.

File: industrial_robot_client/CMakeLists.txt [hunk 1]
```diff
@@ -1,15 +1,13 @@
...
-## System dependencies are found with CMake's conventions
-# find_package(Boost REQUIRED COMPONENTS system)
 find_package(Boost REQUIRED COMPONENTS system thread)
 
 # The definition is copied from the CMakeList for the simple_message package.
-add_definitions(-DROS=1)  #build using ROS libraries
+add_definitions(-DROS=1)           #build using ROS libraries
 add_definitions(-DLINUXSOCKETS=1)  #build using LINUX SOCKETS libraries
 
 set(SRC_FILES src/joint_relay_handler.cpp
```
Comment (gavanderhoorn) [inline industrial_robot_client/CMakeLists.txt:15]: Stray tab at line 14 (rest of file is indented with spaces)

File: industrial_robot_client/CMakeLists.txt [hunk 2]
```diff
@@ -1,15 +1,13 @@
-# http://ros.org/doc/groovy/api/catkin/html/user_guide/supposed.html
 cmake_minimum_required(VERSION 2.8.3)
+
 project(industrial_robot_client)
-# Load catkin and all dependencies required for this package
+
 find_package(catkin REQUIRED COMPONENTS roscpp std_msgs sensor_msgs control_msgs trajectory_msgs simple_message actionlib_msgs actionlib urdf industrial_msgs industrial_utils)
```
Comment (gavanderhoorn) [inline industrial_robot_client/CMakeLists.txt]: Perhaps add some hard linebreaks? For readability?

File: industrial_robot_client/CMakeLists.txt [hunk 3]
```diff
@@ -20,57 +18,13 @@ set(SRC_FILES src/joint_relay_handler.cpp
...
-## DEPENDS: system dependencies of this project that dependent projects also need
-
-# NOTE: The libraries generated this package are not included in the catkin_package
-# macro because libraries must be explicitly linked in projects that depend on this
-# package.  If this is not done (and these libraries were exported), then multiple
-# library definitions (normal - industrial_robot_client and byteswapped - 
-# industrial_robot_client_bswap) are both included (this is bad).
-
 catkin_package(
     CATKIN_DEPENDS roscpp std_msgs sensor_msgs control_msgs trajectory_msgs simple_message actionlib_msgs actionlib urdf industrial_msgs industrial_utils
```
Comment (gavanderhoorn) [inline industrial_robot_client/CMakeLists.txt]: Wrapping again

File: industrial_robot_client/CMakeLists.txt [hunk 4]
```diff
@@ -20,57 +18,13 @@ set(SRC_FILES src/joint_relay_handler.cpp
...
     INCLUDE_DIRS include
 )
```
Comment (gavanderhoorn) [inline industrial_robot_client/CMakeLists.txt:33]: Spaces after `)`

File: industrial_robot_client/CMakeLists.txt [hunk 5]
```diff
@@ -88,14 +42,12 @@ include_directories(include
 #      or
 #    target_link_libraries(my_node industrial_robot_client_bswap)
 
-
```
Comment (gavanderhoorn) [inline industrial_robot_client/CMakeLists.txt:91]: Comment on clarifying text block:
- states there are 2 variants, missing _float64_?
- refers to _rosmake_

File: industrial_robot_simulator/CMakeLists.txt [hunk 6]
```diff
@@ -1,30 +1,13 @@
-# http://ros.org/doc/groovy/api/catkin/html/user_guide/supposed.html
 cmake_minimum_required(VERSION 2.8.3)
+
 project(industrial_robot_simulator)
-# Load catkin and all dependencies required for this package
-find_package(catkin REQUIRED COMPONENTS roscpp std_msgs sensor_msgs control_msgs trajectory_msgs industrial_robot_client)
 
-# Python setup
-#catkin_python_setup()
+find_package(catkin REQUIRED COMPONENTS roscpp std_msgs sensor_msgs control_msgs trajectory_msgs industrial_robot_client)
```
Comment (gavanderhoorn) [inline industrial_robot_simulator/CMakeLists.txt]: line breaks?

**Comment by gavanderhoorn** [comment]
In `industrial_trajectory_filters/CMakeLists.txt`:

```
catkin_package(
    ...
    LIBRARIES ${MOVEIT_LIB_NAME}
)
```

But `${MOVEIT_LIB_NAME}` is never defined. Should probably be `${PROJECT_NAME}`?

**Comment by gavanderhoorn** [comment]
Do we still want to build static libraries in `simple_message`? AFAIK `set(ROS_BUILD_STATIC_LIBS true)` doesn't do that anymore. CMake does this using: `add_library(.. STATIC ..)`.

Same for `industrial_robot_client`.

File: industrial_robot_client/CMakeLists.txt (hunk 5 above)
Comment (shaun-edwards) [inline industrial_robot_client/CMakeLists.txt:91]: Generic clients aren't with float64 support, there are only two variants.

**Comment by shaun-edwards** [comment]
Regarding `${MOVEIT_LIB_NAME}`, fixed.

Regarding `industrial_robot_client`, I'm hesitant to make that change this close to a release.  If it doesn't harm anything, I think we should leave it as is.

**Comment by shaun-edwards** [comment]
@gavanderhoorn, addressed inline issues.  See comments above for others.

File: industrial_msgs/CMakeLists.txt [hunk 7]
```diff
@@ -28,21 +22,10 @@ add_service_files(
...
-## The catkin_package macro generates cmake config files for your package
-## Declare things to be passed to dependent projects
-## LIBRARIES: libraries you create in this project that dependent projects also need
-## CATKIN_DEPENDS: catkin_packages dependent projects also need
-## DEPENDS: system dependencies of this project that dependent projects also need
 catkin_package(
     CATKIN_DEPENDS message_runtime std_msgs trajectory_msgs genmsg
-)
-
+)
```
Comment (gavanderhoorn) [inline industrial_msgs/CMakeLists.txt:31]: nitpicking: missing an empty line.

File: industrial_msgs/CMakeLists.txt [hunk 8]
```diff
@@ -28,21 +22,10 @@ add_service_files(
...
-###################################
-## catkin specific configuration ##
-###################################
-## The catkin_package macro generates cmake config files for your package
-## Declare things to be passed to dependent projects
-## LIBRARIES: libraries you create in this project that dependent projects also need
-## CATKIN_DEPENDS: catkin_packages dependent projects also need
-## DEPENDS: system dependencies of this project that dependent projects also need
 catkin_package(
     CATKIN_DEPENDS message_runtime std_msgs trajectory_msgs genmsg
```
Comment (gavanderhoorn) [inline industrial_msgs/CMakeLists.txt:30]: small: indentation difference

File: industrial_robot_simulator/CMakeLists.txt [hunk 9]
```diff
@@ -33,8 +23,6 @@ catkin_package(
 ## Install ##
 #############
 
-# all install targets should use catkin DESTINATION variables
-# See http://ros.org/doc/groovy/api/catkin/html/adv_user_guide/variables.html
 
 install(PROGRAMS industrial_robot_simulator DESTINATION ${CATKIN_PACKAGE_BIN_DESTINATION})
```
Comment (gavanderhoorn) [inline industrial_robot_simulator/CMakeLists.txt:28]: also missing empty line (not sure)

**Comment by gavanderhoorn** [comment]
Apart from some mixed tabs and spaces in `simple_message/CMakeLists.txt`, this looks fine to me. +1.

**Pull request merged**